src/os_posix/os_union_fs.c

```diff
@@ -833,12 +809,27 @@ __union_fs_file_truncate(WT_FILE_HANDLE *file_handle, WT_SESSION *wt_session, wt
 
     fh = (WT_UNION_FILE_HANDLE *)file_handle;
 
-    // If we truncate a range we'll never need to read that range from the source file. Mark it as
-    // such.
+    /*
+     * If we truncate a range we'll never need to read that range from the source file. Mark it as
+     * such.
+     */
     __union_fs_file_size(file_handle, wt_session, &old_len);
+
+    if (old_len == len)
+        /* Sometimes we call truncate but don't change the length. Ignore */
+        return (0);
+
+    if (len > old_len)
+        /*
+         * It's technically allowed to extend a file by calling truncate. We're ignoring this case
+         * for now.
```

Comment:
The wording "We're ignoring this case for now" is both true but also confusing in that we depend on positive length truncates directly into the posix OS. Can we update this comment in anyway to help with that?

Reply:
I decided to just implement the positive truncate case. Updated in [629ad9c](https://github.com/wiredtiger/wiredtiger/pull/11268/commits/629ad9c2a904cd08ef5a0d68c22a7d490e04cc43)